src/sentry/models/activity.py

```diff
@@ -40,7 +40,13 @@
 
 
 class ActivityManager(BaseManager["Activity"]):
-    def get_activities_for_group(self, group: Group, num: int) -> Sequence[Activity]:
+    def get_activities_for_group(self, group: Group) -> Sequence[Activity]:
+        """
+        Returns a list of Activity objects for the given group, ordered by most recent first,
+        with duplicate consecutive activities (by type, ident, user_id, and data) filtered out,
+        except for notes which are always included. Also appends a synthetic FIRST_SEEN activity
+        at the end, using the group's initial priority if available.
+        """
         activities = []
         activity_qs = self.filter(group=group).order_by("-datetime")
```

Comment:
This does a full table scan. We may want to add a new index.

Reply:
You can get metrics on the query here:
https://sentry.sentry.io/insights/backend/database/spans/span/35fcb71003a016d1/?project=1&span.domain=%2A%2Csentry_activity%2C%2A&statsPeriod=7d